its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java

```diff
@@ -439,6 +452,81 @@ public void test_MarkIssueAs_Dialog() {
   }
+  
+  // integration test for focussing on new code in connected mode
+  @Test
+  public void test_new_code_period_preference() {
+    // 1) create project on server / run first analysis
+    adminWsClient.projects()
+      .create(CreateRequest.builder()
+        .setName(MAVEN_TAINT_PROJECT_KEY)
+        .setKey(MAVEN_TAINT_PROJECT_KEY).build());
+    orchestrator.getServer().associateProjectToQualityProfile(MAVEN_TAINT_PROJECT_KEY, "java", "SonarLint IT New Code");
+    
+    //setNewCodePeriodToDays(MAVEN_TAINT_PROJECT_KEY);
+    
+    orchestrator.executeBuild(MavenBuild.create(new File("projects", "java/maven-taint/pom.xml"))
+      .setCleanPackageSonarGoals()
+      .setProperty("sonar.login", Server.ADMIN_LOGIN)
+      .setProperty("sonar.password", Server.ADMIN_PASSWORD)
+      .setProperty("sonar.projectKey", MAVEN_TAINT_PROJECT_KEY));
+    
+    // 2) import project / check that new code period preference does nothing in standalone mode
+    new JavaPerspective().open();
+    var rootProject = importExistingProjectIntoWorkspace("java/maven-taint", MAVEN_TAINT_PROJECT_KEY);
+
+    var onTheFlyView = new OnTheFlyView();
+    onTheFlyView.open();
+    var taintVulnerabilitiesView = new SonarLintTaintVulnerabilitiesView();
+    taintVulnerabilitiesView.open();
+    
+    openFileAndWaitForAnalysisCompletion(rootProject.getResource("src/main/java", "taint", "taint_issue.java"));
+    
+    await().untilAsserted(() -> assertThat(onTheFlyView.getItems()).hasSize(2));
+    
+    setNewCodePreference(IssuePeriod.NEW_CODE);
+    
+    openFileAndWaitForAnalysisCompletion(rootProject.getResource("src/main/java", "taint", "taint_issue.java"));
+    
+    await().untilAsserted(() -> assertThat(onTheFlyView.getItems()).hasSize(2));
+    
+    setNewCodePreference(IssuePeriod.ALL_TIME);
+    
+    // 3) bind to project on SonarQube / check issues and taint vulnerabilities exist
+    createConnectionAndBindProject(orchestrator, MAVEN_TAINT_PROJECT_KEY, Server.ADMIN_LOGIN, Server.ADMIN_PASSWORD);
+    
+    var notificationShell = new DefaultShell("SonarLint Binding Suggestion");
+    new DefaultLink(notificationShell, "Don't ask again").click();
+    
+    await().untilAsserted(() -> assertThat(onTheFlyView.getItems()).hasSize(2));
+    await().untilAsserted(() -> assertThat(taintVulnerabilitiesView.getItems()).hasSize(1));
+    
+    // 4) unbind project / run second analysis / set new code period since first analysis
+    var bindingsView = new BindingsView();
+    bindingsView.open();
+    bindingsView.removeAllBindings();
+    
+    orchestrator.executeBuild(MavenBuild.create(new File("projects", "java/maven-taint/pom.xml"))
+      .setCleanPackageSonarGoals()
+      .setProperty("sonar.login", Server.ADMIN_LOGIN)
+      .setProperty("sonar.password", Server.ADMIN_PASSWORD)
+      .setProperty("sonar.projectKey", MAVEN_TAINT_PROJECT_KEY));
+    
+    setNewCodePeriodToLastAnalysis(MAVEN_TAINT_PROJECT_KEY);
```

Comment:
Changing the new code period on the server should be done before running the analysis if you want it to be taken into account.